src/bootstrap/config.rs

```diff
@@ -889,26 +891,51 @@ impl Default for RustOptimize {
     }
 }
 
-fn deserialize_and_validate_opt_level<'de, D>(d: D) -> Result<String, D::Error>
+fn deserialize_and_validate_opt_level_string<'de, D>(d: D) -> Result<String, D::Error>
 where
     D: serde::de::Deserializer<'de>,
 {
     let v = String::deserialize(d)?;
-    if ["0", "1", "2", "3", "s", "z"].iter().find(|x| **x == v).is_some() {
+    if ["s", "z"].iter().find(|x| **x == v).is_some() {
         Ok(v)
     } else {
-        Err(format!(r#"unrecognized option for rust optimize: "{}", expected one of "0", "1", "2", "3", "s", "z""#, v)).map_err(serde::de::Error::custom)
+        Err(format!(
+            r#"unrecognized option for rust optimize: "{}", expected one of 0, 1, 2, 3, "s", "z""#,
+            v
+        ))
+        .map_err(serde::de::Error::custom)
+    }
+}
+
+fn deserialize_and_validate_opt_level_int<'de, D>(d: D) -> Result<i8, D::Error>
+where
+    D: serde::de::Deserializer<'de>,
+{
+    let v = i8::deserialize(d)?;
+    if [0, 1, 2, 3].iter().find(|x| **x == v).is_some() {
+        Ok(v)
+    } else {
+        Err(format!(
+            r#"unrecognized option for rust optimize: "{}", expected one of 0, 1, 2, 3, "s", "z""#,
+            v
+        ))
+        .map_err(serde::de::Error::custom)
     }
 }
 
 impl RustOptimize {
     pub(crate) fn is_release(&self) -> bool {
-        if let RustOptimize::Bool(true) | RustOptimize::String(_) = &self { true } else { false }
+        if let RustOptimize::Bool(true) | RustOptimize::String(_) | RustOptimize::Int(_) = &self {
```

Comment:
This is actually pre-existing from the previous PR, but I'm not sure if e.g. `rust.optimize = 0` means that we should automatically enable the release profile? I would expect that for the zero value specifically, we would default to a de bug profile.

Reply:
Good catch! Addressed.